llama_stack/ui/app/chat-playground/page.tsx

```diff
@@ -10,25 +10,546 @@ import {
+            agentId,
+            createdAt: sessionToLoad.started_at
+              ? new Date(sessionToLoad.started_at).getTime()
+              : Date.now(),
+            updatedAt: Date.now(),
+          };
+
+          setCurrentSession(session);
+          console.log(`💾 Saving session ID for agent ${agentId}:`, session.id);
+          SessionUtils.saveCurrentSessionId(session.id, agentId);
+          // cache session data
+          SessionUtils.saveSessionData(agentId, session);
+        } else {
+          // no sessions, create a new one
+          await createDefaultSession(agentId);
+        }
+      } catch (error) {
+        console.error("Error loading agent sessions:", error);
+        // fallback to creating a new session
+        await createDefaultSession(agentId);
+      }
+    },
+    [client, loadSessionMessages, createDefaultSession, selectedModel]
+  );
+
+  useEffect(() => {
+    const fetchAgents = async () => {
+      try {
+        setAgentsLoading(true);
+        const agentList = await client.agents.list();
+        setAgents(
+          (agentList.data as Array<{
+            agent_id: string;
+            agent_config?: {
+              agent_name?: string;
+              name?: string;
+              instructions?: string;
+            };
+            [key: string]: unknown;
+          }>) || []
+        );
+
+        if (agentList.data && agentList.data.length > 0) {
+          // check if there's a previously selected agent
+          const savedAgentId = SessionUtils.loadCurrentAgentId();
+
+          let agentToSelect = agentList.data[0] as {
+            agent_id: string;
+            agent_config?: {
+              agent_name?: string;
+              name?: string;
+              instructions?: string;
+            };
+            [key: string]: unknown;
+          };
+
+          // if we have a saved agent ID, find it in the available agents
+          if (savedAgentId) {
+            const foundAgent = agentList.data.find(
+              (a: { agent_id: string }) => a.agent_id === savedAgentId
+            );
+            if (foundAgent) {
+              agentToSelect = foundAgent as typeof agentToSelect;
+            } else {
+              console.log("Previously slelected agent not found:");
+            }
+          }
+          setSelectedAgentId(agentToSelect.agent_id);
+          SessionUtils.saveCurrentAgentId(agentToSelect.agent_id);
+          // load agent config immediately
+          await loadAgentConfig(agentToSelect.agent_id);
+          // Note: loadAgentSessions will be called after models are loaded
+        }
+      } catch (error) {
+        console.error("Error fetching agents:", error);
+      } finally {
+        setAgentsLoading(false);
+      }
+    };
+
+    fetchAgents();
+
+    // fetch available toolgroups
+    const fetchToolgroups = async () => {
+      try {
+        console.log("Fetching toolgroups...");
+        const toolgroups = await client.toolgroups.list();
```

Comment:
i'm calling toolgroups as I believe this maps to the preferred field in the `AgentConfig`.